### ISSUE TITLE
`download` returns `Path` objects rather than strings when `in_region`

### ISSUE THREAD

**Issue by itcarroll** [nsidc/earthaccess#595]
The [download docs](https://earthaccess.readthedocs.io/en/latest/user-reference/api/api/#earthaccess.api.download) say that the return value shall be a list of strings. When running `in_region` however, the result is a list of `pathlib.PosixPath` objects.

```python
import earthaccess
import pathlib

assert earthaccess.__version__ == "0.9.0"
assert earthaccess.__store__.in_region

local = pathlib.Path("data").mkdir(exist_ok=True)
results = earthaccess.search_data(short_name="PACE_OCI_L2_BGC_NRT", count=1)
paths = earthaccess.download(results, local_path=local)
assert isinstance(paths[0], str)
```

**Comment by mfisher87** [comment]
Nice catch! Why isn't our type checker catching this though?

https://github.com/nsidc/earthaccess/blob/9f43047f6d2f6845d33a9367814262b43d9bcc46/earthaccess/api.py#L177

I love pathlib though, should we update `download()` to always return `Path`s? :)

**Comment by itcarroll** [comment]
If you want to be fancy, you could return the same type given to you in the `local_path` argument.

**Comment by andypbarrett** [comment]
I like the idea of returning a list of Path objects.  I think this would avoid cross platform issues.

**Comment by meteodave** [comment]
I would like to upvote this for `earthaccess.download() ` to be consistently returning an object of the same type, whether it be Posix or string (probably should be Posix for cross platform compatibility although several folks may have developed code already assuming the string list). In my script, I evaluate the `earthaccess.download() ` returned object within my script to check whether I am in region or not and then apply list comprehension line to convert the Posix object to a string list since another function is expecting a string list.

```
files = earthaccess.download(granules, path)
if earthaccess.__store__.in_region == True:  files =  [str(Path(f)) for f in files]       
```

**Comment by Sherwin-14** [comment]
@mfisher87 I want to work on this one.  Should we proceed forward with the suggestion above to return Posix whether in region or out of region?

**Comment by mfisher87** [comment]
Yes, I think we can safely decide to return a `Path` (not a `PosixPath`, some users are on Windows!). We already switched to returning `Path` in some cases, and I think that was driven by a general preference for `Path`s over `str`s. Thanks, Sherwin!

**Comment by asteiker** [comment]
Closing - this was merged via #1031
